src/irmin-graphql/irmin_graphql.ml

```diff
@@ -31,12 +31,40 @@ type commit_input = {
   message: string option;
 }
 
+module Option = struct
+  let map f t = match t with None -> None | Some x -> Some (f x)
+end
+
+module Result = struct
+  let ok x = Ok x
+end
+
 module type CONFIG = sig
   val remote: (?headers:Cohttp.Header.t -> string -> Irmin.remote) option
   val info: ?author:string -> ('a, Format.formatter, unit, Irmin.Info.f) format4 -> 'a
 end
 
-module Make(Server: Cohttp_lwt.S.Server)(Config: CONFIG)(Store : Irmin.S) = struct
+module type PRESENTER = sig
+  type t
+  type src
+
+  val to_src : t -> src
+  val schema_typ : (unit, src option) Schema.typ
+end
+
+module type PRESENTATION = sig
+  module Contents : PRESENTER
+  module Metadata : PRESENTER
+end
+
+module Default_presenter (T : Irmin.Type.S) = struct
+  type t = T.t
+  type src = string
+  let to_src = Irmin.Type.to_string T.t
+  let schema_typ = Schema.string
+end
+
+module MakeExt(Server: Cohttp_lwt.S.Server)(Config: CONFIG)(Store : Irmin.S)(Presentation : PRESENTATION with type Contents.t = Store.contents and type Metadata.t = Store.metadata) = struct
```

Comment:
One minor nit, but I think `MakeExt` should be called `Make_ext` since there is already `Irmin.Make_ext`

Reply:
Ah yes, I always end up camel casing my module names 😞 Fixed!